src/player.cpp

```diff
@@ -2097,14 +2099,28 @@ Item* Player::getCorpse(Creature* lastHitCreature, Creature* mostDamageCreature)
 {
 	Item* corpse = Creature::getCorpse(lastHitCreature, mostDamageCreature);
 	if (corpse && corpse->getContainer()) {
-		std::ostringstream ss;
+		std::unordered_map<std::string, uint16_t> names;
+		for (const auto& killer : getKillers()) {
+			++names[killer->getName()];
+		}
+
 		if (lastHitCreature) {
-			ss << "You recognize " << getNameDescription() << ". " << (getSex() == PLAYERSEX_FEMALE ? "She" : "He") << " was killed by " << lastHitCreature->getNameDescription() << '.';
+			if (!mostDamageCreature) {
+				corpse->setSpecialDescription(fmt::format("You recognize {:s}. {:s} was killed by {:s}{:s}", getNameDescription(), getSex() == PLAYERSEX_FEMALE ? "She" : "He", lastHitCreature->getNameDescription(), names.size() > 1 ? " and others." : "."));
+			} else {
+				if (lastHitCreature != mostDamageCreature && names[lastHitCreature->getName()] == 1) {
```

Comment:
You can collapse the `else` with the `if` here to remove one level of indentation:

```
			if (!mostDamageCreature) {
				...
			} else if (lastHitCreature != mostDamageCreature && names[lastHitCreature->getName()] == 1) {
				...
			} else {
				...
			}
```